src/entrypoint.cpp

```diff
@@ -167,6 +167,56 @@ namespace aperf {
       })
       ->option_text(" ");
 
+    
+
+    struct SubcommandData {
+    std::string metric_command;
+    std::string metric_name;
+    unsigned int frequency;
+    std::string regular_expression;
+    };
+
+    std::vector<SubcommandData> subcommands;
+
+    auto *sub = app.add_subcommand("metric", "For extra metrics, an additional metric" 
+                                  "command can be specified to monitor the profiled" 
+                                  "command. For more details about this option," 
+                                  ": adaptiveperf metric --help")
+                  ->fallthrough()
+                  ->require_subcommand(0, 0);
+
+    std::string metric_cmd = "";
+    sub->add_option("SUBCMD", metric_cmd, "Metric command to be sampled"       
+                  " (required if metric subcommand is called)")
+       ->required()
+       ->check([](const std::string &arg) {
+        if (!arg.empty()) {
+          std::vector<std::string> parts = boost::program_options::split_unix(arg);
+
+          if (parts.empty()) {
+            return "The command you have provided is not a valid one!";
+          }
+        }
+
+        return "";
+      })
+      ->option_text(" ");
+    std::string metric_name = "m";
+    sub->add_option("-N, --metric_name", metric_name, "Name of the sampled metric.");
+    unsigned int metric_freq = 10;
+    sub->add_option("-R,--rate", metric_freq, "Sampling frequency per second for "
+                   "metric command")
+      ->check(OnlyMinRange(1))
+      ->option_text("UINT>0");
+    std::string regular_expression;
+    sub->add_option("-X,--regex", regular_expression, "Regular expression"
+                    "to filter output of the metric command");
+    
+    sub->callback([&subcommands, &metric_cmd, &metric_name, &metric_freq, &regular_expression]() {
+        SubcommandData data = {metric_cmd, metric_name, metric_freq, regular_expression};
+        subcommands.emplace_back(data);
+    });
+    
```

Comment:
I'm not sure if a subcommand is the right way of implementing external metric support in the CLI (command-line interface). It may be confusing for users + your current code doesn't seem to support multiple metrics.